### ISSUE TITLE
Still issues with optional chaining

### ISSUE THREAD

**Issue by neumatho** [meriyah/meriyah#113]
First it looks like it is still needed to enable the **next** option for optional chaining. I think this should not be needed, since it is in the official specification now.

There are still some issues with optional chaining. Try **a?.b[3].c?.(x).d**, then Acorn outputs this tree:

```
{
  "type": "Program",
  "start": 0,
  "end": 16,
  "body": [
    {
      "type": "ExpressionStatement",
      "start": 0,
      "end": 16,
      "expression": {
        "type": "ChainExpression",
        "start": 0,
        "end": 16,
        "expression": {
          "type": "MemberExpression",
          "start": 0,
          "end": 16,
          "object": {
            "type": "CallExpression",
            "start": 0,
            "end": 14,
            "callee": {
              "type": "MemberExpression",
              "start": 0,
              "end": 9,
              "object": {
                "type": "MemberExpression",
                "start": 0,
                "end": 7,
                "object": {
                  "type": "MemberExpression",
                  "start": 0,
                  "end": 4,
                  "object": {
                    "type": "Identifier",
                    "start": 0,
                    "end": 1,
                    "name": "a"
                  },
                  "property": {
                    "type": "Identifier",
                    "start": 3,
                    "end": 4,
                    "name": "b"
                  },
                  "computed": false,
                  "optional": true
                },
                "property": {
                  "type": "Literal",
                  "start": 5,
                  "end": 6,
                  "value": 3,
                  "raw": "3"
                },
                "computed": true,
                "optional": false
              },
              "property": {
                "type": "Identifier",
                "start": 8,
                "end": 9,
                "name": "c"
              },
              "computed": false,
              "optional": false
            },
            "arguments": [
              {
                "type": "Identifier",
                "start": 12,
                "end": 13,
                "name": "x"
              }
            ],
            "optional": true
          },
          "property": {
            "type": "Identifier",
            "start": 15,
            "end": 16,
            "name": "d"
          },
          "computed": false,
          "optional": false
        }
      }
    }
  ],
  "sourceType": "module"
}
```

Meriyah has problems with the [3] and (x) part.

Also the **({})?.a["b"]** parses wrong. It should look like this:

```
{
  "type": "Program",
  "start": 0,
  "end": 12,
  "body": [
    {
      "type": "ExpressionStatement",
      "start": 0,
      "end": 12,
      "expression": {
        "type": "ChainExpression",
        "start": 0,
        "end": 12,
        "expression": {
          "type": "MemberExpression",
          "start": 0,
          "end": 12,
          "object": {
            "type": "MemberExpression",
            "start": 0,
            "end": 7,
            "object": {
              "type": "ObjectExpression",
              "start": 1,
              "end": 3,
              "properties": []
            },
            "property": {
              "type": "Identifier",
              "start": 6,
              "end": 7,
              "name": "a"
            },
            "computed": false,
            "optional": true
          },
          "property": {
            "type": "Literal",
            "start": 8,
            "end": 11,
            "value": "b",
            "raw": "\"b\""
          },
          "computed": true,
          "optional": false
        }
      }
    }
  ],
  "sourceType": "module"
}
```

-Thomas

**Comment by 3cp** [comment]
Thx, #112.

**Comment by 3cp** [comment]
Closed by #112
